src/providers/krb5/krb5_child.c

```diff
@@ -1839,6 +1839,18 @@ static errno_t get_pkinit_identity(TALLOC_CTX *mem_ctx,
         module_name = "p11-kit-proxy.so";
     }
 
+    /* The ':' character is used as a seperator and libkrb5 currently does not
+     * allow to escape it in names. So we have to error out if any of the
+     * names contains a ':' */
+    if (strchr(token_name, ':') != NULL || strchr(module_name, ':') != NULL
+            || strchr(key_id, ':') != NULL || strchr(label, ':') != NULL) {
```

Comment:
Checks below suggest those strings can be NULL.
Is it safe to call `strchr()` in this case?